internal/taskqueue/rabbitmq/rabbitmq.go

```diff
@@ -378,6 +381,18 @@ func redial(ctx context.Context, l *zerolog.Logger, url string) chan chan sessio
 			}
 
 			newSession, err := getSession(ctx, l, url)
+			t.ready = true
+
+			ch := newSession.Connection.NotifyClose(make(chan *amqp.Error, 1))
```

Comment:
This seems to be the recommended way to check if the connection is broken. I tried killing rabbitmq and apart from running in a big amount (infinite?) of unmarshaling errors, this works (so ready is set to false upon killing rabbitmq).